### PR TITLE
Include GitHub Actions

### LINKED ISSUE #97: Add GitHub Actions
We should include github actions in order to ensure that code contributed to the repository adheres to expected formatting and is readable.

An example from @mnlevy1981 : "FEISTY (python-based fish model) uses [pre-commit](https://pre-commit.com/) (configured with [this yaml file](https://github.com/marbl-ecosys/feisty/blob/main/.pre-commit-config.yaml)), and then has a Github Action that [runs the pre-commit tests](https://github.com/mnlevy1981/feisty/blob/main/.github/workflows/linting.yaml)."

### PR DESCRIPTION
**Pull request by TeaganKing**
Closes #97 
This PR uses some of the pre-commit-hooks from [this repository](https://github.com/pre-commit/pre-commit-hooks/blob/main/.pre-commit-config.yaml) and an updated workflow from [here](https://github.com/pre-commit/pre-commit-hooks/blob/main/.github/workflows/main.yml).

### CODE REVIEW

**Comment by TeaganKing** [comment]
Ok pre-commits are now working properly and conflicts between black-jupyter and autopep8, as well as the double vs single quotes, are resolved (thanks to some suggestions from Anissa on the former!)

Anyways, this should be ready for review now.

File: docs/conf.py [hunk 1]
```diff
@@ -3,16 +3,16 @@
...
 # https://www.sphinx-doc.org/en/master/usage/configuration.html
-
 # -- Path setup --------------------------------------------------------------
-
 # If extensions (or modules to document with autodoc) are in another directory,
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
-import os
-import sys
+from __future__ import annotations
```
Comment (mnlevy1981) [inline docs/conf.py:10]: It doesn't look like `annotations` is used anywhere; was this requested by something in the CI testing?

File: examples/nblibrary/ice/plot_diff.py [hunk 2]
```diff
@@ -1,92 +1,100 @@
+from __future__ import annotations
```
Comment (mnlevy1981) [inline examples/nblibrary/ice/plot_diff.py:1]: same comment; I assume it's getting added by the CI but I don't understand why

**Review by mnlevy1981 — CHANGES_REQUESTED** [review]
I'm not sure of the best place for it (maybe the `README` for now, but eventually in a "developer's guide" type page?), but we should include instructions on how to enable `pre-commit` locally. The real power of the package is that it will warn you when you try to commit changes that violate the style guide; using it in Github Actions is more of a backup plan than anything else.

Other than that (and my confusion around `annotate`), this all looks great! Little is as satisfying as seeing a whole bunch of trailing whitespace get removed :)

**Comment by TeaganKing** [comment]
I added a note on running `pre-commit run` to our [contributor's guide](https://github.com/NCAR/CUPiD/wiki/Contributor's-Guide). If you prefer this goes in the readme, I could add it there too, but it seems to fit better in the wiki page (although perhaps it is a little hidden).

I think the `from __future__ import annotations` is needed for checking imports with `reorder-python-imports`, which is a bit annoying.

**Review by mnlevy1981 — APPROVED** [review]
> I added a note on running `pre-commit run` to our [contributor's guide](https://github.com/NCAR/CUPiD/wiki/Contributor's-Guide). If you prefer this goes in the readme, I could add it there too, but it seems to fit better in the wiki page (although perhaps it is a little hidden).

The contributor guide is a great place for it! I didn't think to look there, and the wiki files don't show up as changed in the PR :) I'll come back this afternoon with a small update (adding `precommit` to `environments/dev-environment.yml` and giving more explicit instructions in the wiki about setting it up to run with every `git commit`) but I'll merge this now

**Comment by TeaganKing** [comment]
Sounds great! Thanks for reviewing and merging this, Mike!

**Pull request merged**